MdeModulePkg/Universal/SetupBrowserDxe/IfrParse.c

```diff
@@ -180,14 +180,32 @@ CreateQuestion (
         // Insert to Name/Value varstore list
         //
         NameValueNode = AllocateZeroPool (sizeof (NAME_VALUE_NODE));
-        ASSERT (NameValueNode != NULL);
+        // MU_CHANGE [BEGIN] - CodeQL change
+        if (NameValueNode == NULL) {
+          ASSERT (NameValueNode != NULL);
+          return NULL;
+        }
+
         NameValueNode->Signature = NAME_VALUE_NODE_SIGNATURE;
         NameValueNode->Name      = AllocateCopyPool (StrSize (Statement->VariableName), Statement->VariableName);
-        ASSERT (NameValueNode->Name != NULL);
+        if (NameValueNode->Name == NULL) {

```

Comment:
In this null blocks, you can free prior allocations that succeeded before returning.

Reply:
Updated but please come sanity check my work to make sure I layered the frees and entry list removals correctly

Reply:
Updated